### ISSUE TITLE
Clippy does not respect inline #[allow(…)] attribute

### ISSUE THREAD

**Issue by antoyo** [rust-lang/rust-clippy#2332]
Hello.
It seems that clippy does not respect the inline `#[allow(…)]` attributes.
For instance, the following code still triggers a warning:
```rust
    #[allow(toplevel_ref_arg)]
    let ref mut x = 1234543;
```
Thanks to fix this issue.

**Comment by oli-obk** [comment]
We allow it on some lints. Not sure what's happening with this one. It might be a rustc issue.
